### ISSUE TITLE
Emacs OS X GUI doesn't work

### ISSUE THREAD

**Issue by asmeurer** [conda-forge/emacs-feedstock#2]
I guess it's because we build with `--disable-ns-self-contained`. The GUI is actually unusable for me. It opens behind the terminal, without starting an actual OS X application in the Dock. All keyboard commands are sent to the terminal.

**Comment by asmeurer** [comment]
Attempt to fix at https://github.com/conda-forge/emacs-feedstock/pull/3

**Comment by jakirkham** [comment]
So I saw a message about pulling broken OS X builds, but was on my phone. Does that need to happen? I can put them under a different channel label (e.g. `broken`) so you can still get to them if you need, but they won't get installed by default.

**Comment by asmeurer** [comment]
There's no need to move stuff for now. I've got a "working" build (still some issues, but I'm trying to fix them now).

**Comment by asmeurer** [comment]
I merged https://github.com/conda-forge/emacs-feedstock/pull/3, but now I'm getting this error (and the package is broken)

```
Warning: arch-dependent data dir '/Users/travis/miniconda3/conda-bld/recipe_1481871472196/work/emacs-25.1/nextstep/Emacs.app/Contents/MacOS/libexec/': No such file or directory
Warning: arch-independent data dir '/Users/travis/miniconda3/conda-bld/recipe_1481871472196/work/emacs-25.1/nextstep/Emacs.app/Contents/Resources/etc/': No such file or directory
Warning: Lisp directory '/Users/travis/miniconda3/conda-bld/recipe_1481871472196/work/emacs-25.1/nextstep/Emacs.app/Contents/Resources/lisp': No such file or directory
Error: charsets directory not found:
/Users/travis/miniconda3/conda-bld/recipe_1481871472196/work/emacs-25.1/nextstep/Emacs.app/Contents/Resources/etc/charsets
Emacs will not function correctly without the character map files.
Please check your installation!
```

(in case anyone is seeing this and needs to fix emacs, downgrade to build 4, `conda install emacs=25.1=4`)

**Comment by asmeurer** [comment]
So it's referencing stuff from the work directory. I checked and the path is hardcoded in the binary executable. If we can't figure out how to make it not do this in the build, the only thing I can think of is to rig the work directory to be as long as the long placeholder prefix, and replace it with the real placeholder prefix. That doesn't sound fun, though (unless conda-build lets you set the work directory somehow, we'd have to manually (re)install miniconda to some long prefix path).

Also, does conda build not delete the work directory during the tests? That would have caught the issue right away. CC @msarahan

**Comment by asmeurer** [comment]
The broken package has been removed from conda-forge. It's still at https://anaconda.org/asmeurer/emacs/files (`emacs=25.1=5`, `broken` channel) in case anyone wants to have a look.

**Comment by asmeurer** [comment]
Oh, conda-build does have a `croot` flag, along with `--no-build-id`. It's not really clear from the docs what the build id is when that flag is used, but I believe this should be sufficient to set a long work directory. 

Now my question is, is there a "correct" way to make permanent modifications to the build scripts in a feedstock? If I just modify `.travis.yml` that will get reset on each rerender.

**Comment by asmeurer** [comment]
Oh, `--no-build-id` means it just skips the `recipe_...` subdirectory entirely.

**Comment by asmeurer** [comment]
OK, here's attempt 2 (work in progress): https://github.com/conda-forge/emacs-feedstock/pull/10. I just went ahead and edited the .travis.yml with large warning signs. 

One big problem, if someone rerenders without leaving those changes in the build will NOT fail, because the tests don't pass (because the work directory isn't being deleted). So I should probably work on making the tests fail so that it doesn't accidentally get broken.

**Comment by quartox** [comment]
I am seeing an error for `25.2-1`. Similar to above, but added for transparency:

```
Warning: arch-dependent data dir '/Users/travis/miniconda3/conda-bld/emacs_1504153408821/work/emacs-25.2/nextstep/Emacs.app/Contents/MacOS/libexec/': No such file or directory
Warning: arch-independent data dir '/Users/travis/miniconda3/conda-bld/emacs_1504153408821/work/emacs-25.2/nextstep/Emacs.app/Contents/Resources/etc/': No such file or directory
Warning: Lisp directory '/Users/travis/miniconda3/conda-bld/emacs_1504153408821/work/emacs-25.2/nextstep/Emacs.app/Contents/Resources/lisp': No such file or directory
Error: charsets directory not found:
/Users/travis/miniconda3/conda-bld/emacs_1504153408821/work/emacs-25.2/nextstep/Emacs.app/Contents/Resources/etc/charsets
Emacs will not function correctly without the character map files.
Please check your installation!
```

I was able to downgrade to `25.1-4` which opens but then the gui does not respond and keyboard tries to input on the bash terminal. I am running Sierra version 10.12.6.

**Comment by asmeurer** [comment]
So after #10, the charsets not found error is gone. The GUI works fine if you double click it (or `open ~/anaconda/Emacs.app`), but if you run `emacs`, which is currently symlinked to `Emacs.app/Contents/MacOS/Emacs`, it does what it was doing before, any keys entered go to the terminal, not the GUI.

**Comment by asmeurer** [comment]
[This](https://stackoverflow.com/questions/10171280/how-to-launch-gui-emacs-from-command-line-in-osx) should fix it.

**Comment by asmeurer** [comment]
New PR at https://github.com/conda-forge/emacs-feedstock/pull/19

**Comment by asmeurer** [comment]
It appears to completely work now. If anyone has any issues, let me know (though considering how long this package has been completely broken, I suspect I may be the only user of this package on OS X).
